barretenberg/cpp/src/barretenberg/commitment_schemes/ipa/ipa.hpp

```diff
@@ -664,13 +694,19 @@ template <typename Curve_> class IPA {
      * @param r 
      * @return Fr 
      */
-    static Fr evaluate_challenge_poly(const std::vector<Fr>& u_challenges_inv, Fr r) {
+    static Fr evaluate_challenge_poly(Fr log_poly_length, const std::vector<Fr>& u_challenges_inv, Fr r) {
+        using Builder = typename Curve::Builder;
+
+        Builder* builder = r.get_context();
         Fr challenge_poly_eval = 1;
         Fr r_pow = r;
-        size_t len = u_challenges_inv.size();
-        for (size_t i = 0; i < len; i++) {
-            challenge_poly_eval *= (Fr(1) + u_challenges_inv[len - 1 - i] * r_pow);
-            r_pow *= r_pow;
+        for (size_t i = 0; i < CONST_ECCVM_LOG_N; i++) {
+            stdlib::bool_t<typename Curve::Builder> dummy_round = stdlib::witness_t(builder, i < CONST_ECCVM_LOG_N - uint32_t(log_poly_length.get_value()));
```

Comment:
this sort of stuff is not secure. Will need to figure out a way to fix it, like the rest of the dummy_round stuff in sumcheck